src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -21,7 +21,27 @@ def get_argument_infrastructure_resource_group(self):
         return self.get_param("infrastructure_resource_group")
 
     def construct_payload(self):
-        super().construct_payload()
+        ### copy from the parent construct_payload
+        self.set_up_app_log_configuration()
+
+        self.managed_env_def["location"] = self.get_argument_location()
+        self.managed_env_def["tags"] = self.get_argument_tags()
+        self.managed_env_def["properties"]["zoneRedundant"] = self.get_argument_zone_redundant()
+
+        self.set_up_workload_profiles()
+
+        if self.get_argument_instrumentation_key() is not None:
+            self.managed_env_def["properties"]["daprAIInstrumentationKey"] = self.get_argument_instrumentation_key()
+
+        # Vnet
+        self.set_up_vnet_configuration()
+
+        if self.get_argument_mtls_enabled() is not None:
+            safe_set(self.managed_env_def, "properties", "peerAuthentication", "mtls", "enabled", value=self.get_argument_mtls_enabled())
+        ### copy end
```

Comment:
why not call parent construct_payload directly

Reply:
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/containerapp/containerapp_env_decorator.py#L173-L179

because it hardcode the custom domain logic, we could use this logic